design/cassandra/sidecar.md

```diff
@@ -0,0 +1,38 @@
+## Sidecar Design Proposal
+
+### Control-Loop Design
+
+* Each sidecar will:
+    * Act as a mini-controller for an instance of the Database.
+    * Have a corresponding `Spec` and will update a corresponding `Status`.
+* The controller will create the sidecar's `Spec` and the sidecar will conform to it and update its `Status`
+
+### Examples
+
+Let's consider the case of creating a new Cassandra Cluster. It will look something like this:
+
+1. *User* creates CRD for a Cassandra Cluster.
+2. *Controller* sees the newly created CRD object and creates a StatefulSet for each Cassandra Rack and a clusterIP service for each Pod. Also creates the objects for the sidecars, which will look something like this:
+```yaml
+apiVersion: "cassandra.rook.io/v1alpha1"
+kind: "Sidecar"
```

Comment:
I think the intention here is that this is a `CassandraInstance` or `CassandraDaemon`. I think `Sidecar` is how its implemented not what the Resource is.

Reply:
I have discussed about it with @vkoukis, we are working together as part of my internship in @arrikto. He made a similar comment it would be better to name the resource something like "Member" or "CassandraMember". Since you also made the same remark, I guess it would be more accurate to use a different description. This also brings me to another question:

Is it preferable to name the resource "Cluster" and "Member" and distinguish them from the apiVersion, or is it better to name them "CassandraCluster" and "CassandraMember" ? In the first case, `kubectl get clusters` will not work in a namespace with other cluster objects.